src/debug.rs

```diff
@@ -0,0 +1,356 @@
+use crate::{
+    fork::ForkSource,
+    node::{InMemoryNodeInner, MAX_TX_SIZE},
+};
+use jsonrpc_core::{BoxFuture, Result};
+use once_cell::sync::OnceCell;
+use std::sync::{Arc, RwLock};
+use vm::{
+    constants::ETH_CALL_GAS_LIMIT, CallTracer, ExecutionResult, HistoryDisabled, TxExecutionMode,
+    Vm,
+};
+use zksync_basic_types::H256;
+use zksync_core::api_server::web3::backend_jsonrpc::{
+    error::into_jsrpc_error, namespaces::debug::DebugNamespaceT,
+};
+use zksync_state::StorageView;
+use zksync_types::{
+    api::{BlockId, BlockNumber, DebugCall, DebugCallType, ResultDebugCall, TracerConfig},
+    l2::L2Tx,
+    transaction_request::CallRequest,
+    PackedEthSignature, Transaction, CONTRACT_DEPLOYER_ADDRESS,
+};
+use zksync_web3_decl::error::Web3Error;
+
+/// Implementation of DebugNamespaceImpl
+pub struct DebugNamespaceImpl<S> {
+    node: Arc<RwLock<InMemoryNodeInner<S>>>,
+}
+
+impl<S> DebugNamespaceImpl<S> {
+    /// Creates a new `Debug` instance with the given `node`.
+    pub fn new(node: Arc<RwLock<InMemoryNodeInner<S>>>) -> Self {
+        Self { node }
+    }
+}
+
+impl<S: Send + Sync + 'static + ForkSource + std::fmt::Debug> DebugNamespaceT
+    for DebugNamespaceImpl<S>
+{
+    fn trace_block_by_number(
+        &self,
+        _block: BlockNumber,
+        _options: Option<TracerConfig>,
+    ) -> BoxFuture<Result<Vec<ResultDebugCall>>> {
+        todo!()
```

Comment:
Avoid `todo!()` as this will panic and crash the node. Please use similar method as `not_implemented` in `node.rs` (maybe move that to `utils.rs`)

Reply:
done, also replaced usage of `not_implemented!()` macro in `zks.rs`